certbot/crypto_util.py

```diff
@@ -242,6 +244,100 @@ def valid_privkey(privkey):
         return False
 
 
+def verify_renewable_cert_sig(renewable_cert):
+    """ Verifies the signature of a `storage.RenewableCert` object.
+    :param typ: `storage.RenewableCert`
+
+    :raises OpenSSL.crypto.Error if signature verification fails
+    """
+    try:
+        with open(renewable_cert.chain, 'rb') as chain:
+            with open(renewable_cert.cert, 'rb') as cert:
+                chain, _ = pyopenssl_load_certificate(chain.read())
+                cert = x509.load_pem_x509_certificate(cert.read(), default_backend())
+                hash_name = cert.signature_hash_algorithm.name
+                return OpenSSL.crypto.verify(chain, cert.signature,
+                        cert.tbs_certificate_bytes, hash_name)
+    except (IOError, ValueError, OpenSSL.crypto.Error) as e:
+        error_str = "verifying the signature of the cert located at {0} has failed. \
+                Details: {1}".format(renewable_cert.cert, e)
+        logger.exception(error_str)
+        raise e
+
+def verify_cert_matches_priv_key(renewable_cert):
+    """Do the private key and cert match?
+
+    :param typ: `storage.RenewableCert`
+
+    :raises OpenSSL.SSL.Error if they don't match
+    """
+    try:
+        with open(renewable_cert.cert) as cert:
+            with open(renewable_cert.privkey) as privkey:
+                privkey = OpenSSL.crypto.load_privatekey(
+                        OpenSSL.crypto.FILETYPE_PEM, privkey.read())
+                cert = OpenSSL.crypto.load_certificate(
+                        OpenSSL.crypto.FILETYPE_PEM, cert.read())
+                context = OpenSSL.SSL.Context(OpenSSL.SSL.SSLv23_METHOD)
+                context.use_privatekey(privkey)
+                context.use_certificate(cert)
+                context.check_privatekey()
+    except OpenSSL.SSL.Error as e:
+        error_str = "verifying the cert located at {0} matches the \
+                private key located at {1} has failed. \
+                Details: {2}".format(renewable_cert.cert,
+                        renewable_cert.privkey, e)
+        logger.exception(error_str)
+        raise e
```

Comment:
It's worth considering raising our own error type here for convenience in `verify_renewable_cert`. This is especially true if we start catching `IOError` here as well.